src/orion/core/io/experiment_builder.py

```diff
@@ -200,34 +208,76 @@ def _instantiate_space(config):
     return SpaceBuilder().build(config)
 
 
-def _instantiate_algo(space, max_trials, config=None, ignore_unavailable=False):
+def _instantiate_knowledge_base(kb_config: dict[str, Any]) -> KnowledgeBase:
+    """Instantiate the Knowledge base from its configuration."""
+    if len(kb_config) != 1:
+        raise ConfigurationError(
+            f"The configuration for the KB should only have one key (the name of the KB "
+            f"class) and the dict of kwargs. (got {kb_config})"
+        )
+    kb_type_name = list(kb_config.keys())[0]
+    kb_types_with_name = [
+        subclass
+        for subclass in KnowledgeBase.__subclasses__()
+        if subclass.__qualname__ == kb_type_name
+    ]
+    if len(kb_types_with_name) == 0:
+        raise RuntimeError(
+            f"Unable to find a subclass of KnowledgeBase with the given name: {kb_type_name}"
+        )
+    if len(kb_types_with_name) > 1:
+        raise RuntimeError(
+            f"Multiple subclasses of KnowledgeBase with the given name: {kb_type_name}"
+        )
+    kb_type = kb_types_with_name[0]
+    kb_kwargs = kb_config[kb_type_name]
+    return kb_type(**kb_kwargs)
+
+
+def _instantiate_algo(
+    space: Space,
+    max_trials: int | None,
+    config: type[BaseAlgorithm] | str | dict | None = None,
+    ignore_unavailable: bool = False,
+    knowledge_base: KnowledgeBase | None = None,
+):
     """Instantiate the algorithm object
 
     Parameters
     ----------
-    config: dict, optional
+    config:
         Configuration of the algorithm. If None or empty, system's defaults are used
         (orion.core.config.experiment.algorithms).
     ignore_unavailable: bool, optional
         If True and algorithm is not available (plugin not installed), return the configuration.
         Otherwise, raise Factory error.
 
     """
-    if not config:
-        config = orion.core.config.experiment.algorithms
+    config = config or orion.core.config.experiment.algorithms
 
     try:
-        backported_config = backward.port_algo_config(config)
-        algo_constructor: type[BaseAlgorithm] = algo_factory.get_class(
-            backported_config.pop("of_type")
-        )
-        # NOTE: the config doesn't have the `of_type` key anymore, it only has the algo's kwargs
+        if isinstance(config, str):
+            algo_type = algo_factory.get_class(config)
+            algo_config = {}
+        elif inspect.isclass(config) and issubclass(config, BaseAlgorithm):
```

Comment:
Following up on https://github.com/Epistimio/orion/pull/933/files#r933572718, this elif and the last else clause are likely not covered by the tests. If you find tests covering these could you please share the links here. Thanks!

Reply:
Should be fixed in 5040c07f67c31da2d7bac6a74cfec256d2ae574a

Reply:
Great thanks! There's only the invalid config case missing.

Reply:
Actually, now that I think about it, this last "invalid config" case isn't serving any real purpose, since every declared type for `config` is already covered. Since this is an 'internal' function, I'd argue that this case can be removed. This is also a good use-case for asserts, and this saves us from having to write a useless test.

Reply:
5a61b5988c4ca8bcc66ee0586a0307511999b57c

Reply:
I read the message of that comment with an italian accent, for some reason :)